driver/src/main/java/org/neo4j/driver/exceptions/RoutingException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2002-2019 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.exceptions;
+
+/**
+ * A routing error indicate a fatal problem to obtain routing tables such as the routing table for a specified database does not exist.
+ * This exception should not be retried.
+ * @since 2.0
+ */
+public class RoutingException extends ClientException
```

Comment:
I'm not sure whether we should name this as `RoutingException`.

We're using it either to wrap `Neo.ClientError.Database.DatabaseNotFound` or to signal that the user is connecting to a pre-4.0 server. The only common thing between these two conditions is that the discovery process should be terminated immediately.

Anyways, still not sure - would `FatalDiscoveryException` be more descriptive?